cmd/geth/main.go

```diff
@@ -425,8 +425,10 @@ func console(ctx *cli.Context) {
 	startNode(ctx, node)
 
 	// Attach to the newly started node, and either execute script or become interactive
-	client := utils.NewInProcRPCClient(node)
-
+	client, err := utils.NewRemoteRPCClientFromString("ipc:" + node.IpcEndpoint())
+	if err != nil {
+		utils.Fatalf("Failed to attach to the inproc geth: %v", err)
```

Comment:
Seems I've misunderstood in the inproc implementation. For some reason I thought it was HTTP based. However, this could then itself be a very useful addition to the `rpc` package and also exposed via the `node`, to allow interfacing the API via libraries without the need for any external channels. I'll move it to the `rpc` package and work it into the rest of the code.